### PR TITLE
Draft: Wrote test to check TXT results with default golang LookupTXT

### PR DESCRIPTION
**Pull request by Miniwoffer**
tests for: https://github.com/domainr/dnsr/issues/111

### CODE REVIEW

File: resolver_test.go [hunk 1]
```diff
@@ -12,6 +12,29 @@ import (
...
+				exsists = true
+			}
+		}
+		if !exsists {
+			t.Errorf("TXT record %q not found", rr)
+		}
+	}
+	if count(rrs, func(rr RR) bool { return rr.Type == "TXT" }) != len(rrs2) {
+		t.Errorf("TXT record count mismatch: %d != %d", count(rrs, func(rr RR) bool { return rr.Type == "TXT" }), len(rrs2))
+	}
```
Comment (Integralist) [inline resolver_test.go:35]: nit: dedupe count execution

```suggestion
	c := count(rrs, func(rr RR) bool { return rr.Type == "TXT" })
	if c != len(rrs2) {
		t.Errorf("TXT record count mismatch: %d != %d", c, len(rrs2))
	}
```

File: resolver_test.go [hunk 2]
```diff
@@ -12,6 +12,29 @@ import (
...
+	rrs2, err := net.LookupTXT(domain)
+	st.Expect(t, err, nil)
+	for _, rr := range rrs2 {
+		exsists := false
+		for _, rr2 := range rrs {
+			if rr2.Type == "TXT" && rr == rr2.Value {
+				exsists = true
+			}
+		}
+		if !exsists {
```
Comment (Integralist) [inline resolver_test.go:29]: fix typo

```suggestion
		exists := false
		for _, rr2 := range rrs {
			if rr2.Type == "TXT" && rr == rr2.Value {
				exists = true
			}
		}
		if !exists {
```

**Review by Integralist — COMMENTED** [review]
👋🏻 @Miniwoffer  

Thanks for opening this PR!

I've made a couple of minor suggested edits.

@case @cee-dub @ydnar is this a PR you would be happy to accept? I ask because the benefit of comparing the results from `ResolveErr` to `net.LookupTXT` is unclear to me (other than proving the dnsr result aligns with Go's default lookup implementation).

**Comment by Integralist** [comment]
Looks like the `TestGoogleTXT` test fails:

```
=== RUN   TestGoogleTXT
    resolver_test.go:30: TXT record "globalsign-smime-dv=CDYX+XFHUw2wml6/Gb8+59BsH31KzUr6c1l2BPvqKX8=" not found
    resolver_test.go:30: TXT record "google-site-verification=wD8N7i1JTNTkezJ49swvWW48f8_9xveREV4oB-0Hf5o" not found
    resolver_test.go:30: TXT record "webexdomainverification.8YX6G=6e6922db-e3e6-4a36-904e-a805c28087fa" not found
    resolver_test.go:30: TXT record "MS=E4A68B9AB2BB9670BCE1[54](https://github.com/domainr/dnsr/actions/runs/7421023746/job/20194239527?pr=112#step:6:55)12F62916164C0B20BB" not found
    resolver_test.go:30: TXT record "v=spf1 include:_spf.google.com ~all" not found
    resolver_test.go:30: TXT record "google-site-verification=TV9-DBe4R80X4v0M4U_bd_J9cpOJM0nikft0jAgjmsQ" not found
    resolver_test.go:34: TXT record count mismatch: 6 != 12
```

Which would make sense due to the original logic comment...

```
// Google will have at least an SPF record, but might transiently have verification records too.
```

This would be why it uses `>=` instead.

**Comment by Miniwoffer** [comment]
Yeah, this was created in reference to https://github.com/domainr/dnsr/issues/111, its suppose to fail.

**Comment by Miniwoffer** [comment]
Should probably have tagged it as Draft or something.

**Comment by Integralist** [comment]
That's OK no worries, I'll move it to a Draft PR for now. Thanks!

**Comment by Miniwoffer** [comment]
@Integralist 
Make another MR for the fix #113.
I could make these tests a little more race-condition-safe by retrying a few times on failure, but if you feel like its an unnecessary test you can just close it.

**Comment by cee-dub** [comment]
Superseded by #120

**Pull request closed without merging**